dora/core/server/master/src/main/java/alluxio/master/job/DoraLoadJob.java

```diff
@@ -108,11 +108,14 @@ public class DoraLoadJob extends AbstractJob<DoraLoadJob.DoraLoadTask> {
 
   // Job states
   private final Queue<String> mRetryFiles = new ArrayDeque<>();
+  private final Map<String, Integer> mRetryCount = new ConcurrentHashMap<>();
```

Comment:
I don't quite understand the meaning of retry count since if it's retryable we just retry and should not retry a lot of times, and if it's not retryable we just do not retry. What we should do is pass the correct exception to the scheduler

Reply:
so first this retry-able flag isn't reliable... Workers just populate this flag based on exception type and sometimes we just don't handle this correctly. E.g. in the previous implementation, any UnknownRuntimeException will be marked as not retryable, which is pretty much wrong. Plus there's individual UFS implementations which may throw different exceptions. It's hard to give a correct flag on the worker side.

secondly, even if an error is retry-able, we should not retry it indefinitely.... That endless retry might block the following tasks or decrease the performance significantly. 

also sometimes the retry-able flag is not available. What if the RPC just fails? Do we also want to retry it here?

Reply:
https://www.alibabacloud.com/help/en/oss/developer-reference/error-handling-1
Regarding the error handling, we can polish our UFS exceptions like AlluxioS3Exception and handle each exception properly. I agree that we could have a retry limit for failed files to avoid unlimited retry, But it should be combined with the health threshold to ensure we are not storing too many failed files.